staging/src/k8s.io/controller-manager/app/helper.go

```diff
@@ -29,20 +29,20 @@ import (
 )
 
 // WaitForAPIServer waits for the API Server's /healthz endpoint to report "ok" with timeout.
-func WaitForAPIServer(client clientset.Interface, timeout time.Duration) error {
+func WaitForAPIServer(ctx context.Context, client clientset.Interface, timeout time.Duration) error {
 	var lastErr error
 
 	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
```

Comment:
Use `wait.PollImmediateWithContext(ctx, time.Second, timeout, func(ctx context.Context) (bool, error) {` here.

The advantage is that if ctx has a timeout or deadline, polling will stop when reaching it.